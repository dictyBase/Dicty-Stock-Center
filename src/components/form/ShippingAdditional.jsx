import React, { Component, PropTypes } from 'react'
import Panel from 'dicty-react-components/src/Panel'
import PanelHeader from 'dicty-react-components/src/PanelHeader'
import PanelTitle from 'dicty-react-components/src/PanelTitle'
import PanelBody from 'dicty-react-components/src/PanelBody'
import Comments from './Comments'
import 'styles/core.scss'

export default class ShippingAdditional extends Component {
    displayName = 'shipping details';

    static propTypes = {
        comments: PropTypes.object.isRequired,
        title: PropTypes.string.isRequired
    }

    render() {
        const { title, comments } = this.props
        const panelStyle = { border: '1px solid #D2D7D3' }
        const headerStyle = {
            backgroundColor: '#337ab7',
            padding: '20px',
            borderColor: '#4B77BE'
        }
        const titleStyle = {
            color: '#ffffff',
            fontWeight: '200',
            fontSize: '20px'
        }
        return (
            <Panel style={ panelStyle }>
                <PanelHeader style={ headerStyle }>
                   <PanelTitle style={ titleStyle }>{ title }</PanelTitle>
                </PanelHeader>
                <PanelBody>
                    <Comments comments= { comments } rows = { '5' }
                        placeholder = { 'Please enter any comments or special instructions here' }>
                        Comments:
                    </Comments>
                </PanelBody>
            </Panel>
        )
    }
}
