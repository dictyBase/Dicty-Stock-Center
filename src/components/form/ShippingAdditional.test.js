import React from 'react'
import renderer from 'react-test-renderer'
import ShippingAdditional from './ShippingAdditional'

test('matching a snapshot of ShippingAdditional', () => {
    const comments = {
        value: undefined
    }
    const placeholder = 'placeholder'
    const rows = '20'
    const title = 'Testing'

    const component = renderer.create(
    <ShippingAdditional
        rows={ rows }
        comments={ comments }
        placeholder={ placeholder }
        title={ title }
        />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
