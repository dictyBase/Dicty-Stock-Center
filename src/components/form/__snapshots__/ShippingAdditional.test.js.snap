// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ShippingAdditional 1`] = `
<div
  className="gsdTnU"
  style={
    Object {
      "border": "1px solid #D2D7D3",
      "marginBottom": 0,
    }
  }
>
  <div
    className="bBgvkp"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#337ab7",
        "borderBottom": "0px none #ddd",
        "borderColor": "#4B77BE",
        "padding": "20px",
      }
    }
  >
    <h3
      className="jXCrp"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "20px",
          "fontWeight": "200",
        }
      }
    >
      <div
        className="kyzudI"
        onClick={[Function]}
      >
        Testing
      </div>
    </h3>
  </div>
  <div
    className="panel-body cgBkWQ"
    open={true}
    style={Object {}}
  >
    <div
      className="DRXhs"
    />
    <div
      className="sc-hmXxxW ggtKDs"
    >
      <div
        className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
          is={null}
        >
          <label
            className="sc-ipXKqB hIQKmZ"
          >
            Comments:
          </label>
        </div>
        <div
          className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
          is={null}
        >
          <textarea
            className="sc-qrIAp kwjHFv"
            placeholder="Please enter any comments or special instructions here"
            rows="5"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className="DRXhs"
    />
  </div>
</div>
`;
